packages/vm/src/parsed_wasm.rs

```diff
@@ -179,6 +182,17 @@ impl<'a> ParsedWasm<'a> {
                 Payload::ExportSection(e) => {
                     this.exports = e.into_iter().collect::<Result<Vec<_>, _>>()?;
                 }
+                Payload::CustomSection(reader) if reader.name() == "cw_state_version" => {
+                    // This is supposed to be valid UTF-8
+                    let raw_version = str::from_utf8(reader.data())
+                        .map_err(|err| VmError::parse_err("str", err))?;
```

Comment:
I think we better make this a `VmError::static_validation_err` to provide more context what is happening here